src/libsyntax/parse/parser/pat.rs

```diff
@@ -384,24 +384,85 @@ impl<'a> Parser<'a> {
         })
     }
 
+    fn parse_pat_ident_mut(&mut self) -> PResult<'a, PatKind> {
+        let mut_span = self.prev_span;
+
+        if self.eat_keyword(kw::Ref) {
+            return self.recover_mut_ref_ident(mut_span)
+        }
+
+        self.recover_additional_muts();
+
+        let mut pat = self.parse_pat(Some("identifier"))?;
```

Comment:
Hmm; I realized now that this approach will also allow `let mut $i` where `$i:pat` whereas it was not allowed before which would be a bug... We should be able to `.check_ident()` instead and in the `else` branch we can unconditionally error and avoid this bug...

Reply:
Fixed in dbbe3363c94b120d1eba9cba01dadddd862716b8, I think.